src/File.jl

```diff
@@ -37,9 +40,27 @@ function File(
     path::AbstractString;
     load_metadata::Bool=true,
     require_modality::Bool=true,
-    strict::Bool=true
+    strict::Bool=true,
+    extract_from_full_path::Bool=true
 )
     entities = parse_path(path, require_modality=require_modality, strict=strict)
+    # Try extracting sub and ses from full path if not exists in parsed filename
+    if extract_from_full_path
+        entities["sub"] = haskey(entities, "sub") ? entities["sub"] :
```

Comment:
A minor thing: I believe this may be cleaner as simply

```julia
if !haskey(entities, "sub")
    entities["sub"] = ...
end
```

given that it's kinda odd to reassign `entities["sub"] = entities["sub"]` in the case that it already exists.

Reply:
Fair enough! Will update this.